CHANGELOG.md

```diff
@@ -5,6 +5,10 @@
 - feat: add support for custom headers per request
 - feat: add contextual tuples support in Expand requests
 
+> [!WARNING]
+> BREAKING CHANGES:
+> This unreleased version contains a breaking change around `client.RequestOptions` - no longer accepting `MinRetry` and `MaxRetry` fields - they will be re-added before the next release.
+
```

Comment:
Isn't it also a breaking change that `ClientRequestOptions` was renamed to `RequestOptions`? I don't fully understand the relationship between the per-request headers addition and the change from `ClientRequestOptions` -> `RequestOptions`, and how the retry params fit into that. There's clearly a story to tell here and a plan, but it's not clear to me what that is.

Reply:
RequestOptions are all the overrides that apply to requests, headers and retry params, others will follow later (e.g. connection options).

ClientRequestOptions is not a breaking change as it is not used